### DISCUSSION TITLE
jsbach piControl landuseTransitions zero versus 1850

### DISCUSSION THREAD

**Discussion by chrisdane** [esm-tools/esm_tools#695] [Models]
Hi

Jsbach needs two forcing files related to land use changes if dynamic vegetation is enabled: `landuseHarvest.<year>.nc` and `landuseTransitions.<year>.nc`. Whether the latter is zero or not affects the modeled global carbon budget.

# piControl global carbon budget 
For the piControl experiment the two forcing files are currently (esm_tools release) defined via 
https://github.com/esm-tools/esm_tools/blob/1b4d746f8db75a20f6a8fd4f3a1d281b73cf5544/configs/components/jsbach/jsbach.yaml#L110-L116

where LU refers to the Harvest file and LU_trans to the Transitions file:
https://github.com/esm-tools/esm_tools/blob/1b4d746f8db75a20f6a8fd4f3a1d281b73cf5544/configs/components/jsbach/jsbach.datasets.yaml#L12-L20

Finally, these two forcing files will be used (levante):
```
/pool/data/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_harvest/LUH_harvest_T63_1850.nc
/pool/data/JSBACH/input/r0010/T63/New_Hampshire_LCC/no_LUH_transitions_T63.nc 
```
where the harvest file has non-zero entries but the transitions file is zero everywhere. Similarly as for the harvest forcing there is a non-zero file for the transitions for the year 1850:
```
/pool/data/JSBACH/input/r0010/T63/New_Hampshire_LCC/hist_dynveg/LUH_scaled_transitions_T63_1850.nc
```

The globally integrated CO2 flux (expressed as carbon flux) from land use changes (echam variable `co2_flx_lcc`) differs with respect to the used transitions forcing file (black = zero transitions, red = 1850 transitions):
![co2_flx_lcc_awi_esm_1_1_lr_kh800_piControl_vs_awi_esm_1_1_lr_kh800_LUtrans1850_Jan-Dec_980_vs_1002_to_1051_vs_1048_global_samevars_ts](https://user-images.githubusercontent.com/26486859/173795535-e5d9dfd6-40bc-4676-a8cf-130a9ad2af12.png)

This carbon contribution from land use changes is so large that it completely changes the overall carbon flux from land to atmosphere (= nbp = `co2_flx_land` + `co2_flx_lcc` + `co2_flx_harvest`; 50-year mean):
![nbp_awi_esm_1_1_lr_kh800_piControl_vs_awi_esm_1_1_lr_kh800_LUtrans1850_Jan-Dec_1002_to_1048_global_samevars_timmean](https://user-images.githubusercontent.com/26486859/173797103-c2db20ea-aa9b-4113-a99f-438044497b02.png)

With the zero transitions forcing, the land is a carbon sink (nbp > 0). With the non-zero transitions, the land becomes a carbon source. This changes the global carbon budget, i.e. the sum of the land and ocean carbon fluxes (= nbp + `co2_flx_ocean`; 50-year mean):
![co2_flx_total_awi_esm_1_1_lr_kh800_piControl_vs_awi_esm_1_1_lr_kh800_LUtrans1850_Jan-Dec_1002_to_1048_global_samevars_timmean](https://user-images.githubusercontent.com/26486859/173798024-a6e5503c-de2d-4773-a630-a1db071570b9.png)

This change of the global carbon budget from a large land/ocean sink to a small land/ocean source is important for the carbon cycle: Using zero transitions, the large land and ocean sinks constantly draw CO2 from the atmosphere. This leads to unrealistically low atmospheric CO2 concentrations in esm-* experiments (e.g. esm-piControl), in which the CO2 concentration is not prescribed but calculated by the coupled model:
![aCO2_esm_piControl_restartall_Jan-Dec_1_to_287_global_annual](https://user-images.githubusercontent.com/26486859/173800386-ab105727-717b-4ea6-99c9-37960e487b5f.png)

Ideally, the atmospheric CO2 concentration should stay around 283 ppm (or whatever PI value). Using the 1850 transitions file yields a much more stable atmospheric CO2 concentration:
![awicm-1 0-recom_esm-piControl_fesom_aCO2_select](https://user-images.githubusercontent.com/26486859/173801393-884f0998-16e6-45c8-b1c6-17aa4b96a929.png)

(The weird large CO2 concentrations in the first time step come from echams CO2 restart file.) 

# jsbach

Jsbach reads the landuseTransition forcing when `UseLanduseTransitions` is true:
https://gitlab.dkrz.de/modular_esm/echam6/-/blob/master/src/src_jsbach3/mo_jsbach.f90#L357:
```fortran
       SELECT CASE(TRIM(lcc_forcing_type))
       CASE ('NONE')
          options%UseExternalLandcoverMaps = .FALSE.
          options%UseLanduseTransitions    = .FALSE.
          options%UseNetLanduseTransitions = .FALSE.
          CALL message('jsbach_config','No landuse change')
       CASE ('MAPS')
          options%UseExternalLandcoverMaps = .TRUE.
          options%UseLanduseTransitions    = .FALSE.
          options%UseNetLanduseTransitions = .FALSE.
          CALL message('jsbach_config','Calculating landuse change from land cover maps')
       CASE ('TRANSITIONS')
          options%UseExternalLandcoverMaps = .FALSE.
          options%UseLanduseTransitions    = .TRUE.
          options%UseNetLanduseTransitions = .FALSE.
          CALL message('jsbach_config','Calculating landuse change from LU transition maps')
       CASE ('NET_TRANSITIONS')
          options%UseExternalLandcoverMaps = .FALSE.
          options%UseLanduseTransitions    = .TRUE.
          options%UseNetLanduseTransitions = .TRUE.
          CALL message('jsbach_config','Calculating landuse change from LU transition maps but using minimised (net) transitions')
       CASE default
          CALL finish('jsbach_config','Wrong lcc_forcing_type (allowed are: NONE, MAPS, or TRANSITIONS)')
       END SELECT
```
i.e. if `lcc_forcing_type` from the namelist.jsbach is `TRANSITIONS` or `NET_TRANSITIONS`. For piControl and dynveg, this is the case:
https://github.com/esm-tools/esm_tools/blob/1b4d746f8db75a20f6a8fd4f3a1d281b73cf5544/namelists/jsbach/3.20/PI-CTRL/dynveg/namelist.jsbach#L15 

For the non-dynveg case, `lcc_forcing_type` is not defined in namelist.jsbach and the default value is used:
https://gitlab.dkrz.de/modular_esm/echam6/-/blob/master/src/src_jsbach3/mo_jsbach.f90#L224:
```fortran
       lcc_forcing_type = 'NONE'
```
, i.e. not reading the landuseTransition file.

# mpi-esm

The decision to use the zero transitions file was probably based on the choice of the mpi-esm model. The default piControl config of the latest mpiesm tarball [mpiesm-1.2.01p5.tar.gz](https://code.mpimet.mpg.de/projects/mpi-esm-users/files) reads
```
mpiesm-1.2.01p5/run/standard_experiments/piControl-LR.config:
[files]
  [[jsbach]]
    [[[mapped]]]
      [[[[landuse]]]]
        landuseTransitions.%year.nc = no_LUH_transitions_$${ATM_RES}.nc
        landuseHarvest.%year.nc = hist_harvest/LUH_harvest_$${ATM_RES}_$LAND_REF_YEAR.nc
```

# esm_tools

The necessary changes in the esm_tools are simple and available in the [LU_trans_1850_v6](https://github.com/esm-tools/esm_tools/compare/release...LU_trans_1850_v6) branch:
```diff
configs/components/jsbach/jsbach.yaml:
                choose_echam.scenario:
                        PI-CTRL:
                                 forcing_files:
-                                        LU_trans: LU_trans_none
+                                        LU_trans: LU_trans_1850
```
and
```diff
configs/components/jsbach/jsbach.datasets.yaml:
forcing_sources:
+        "LU_trans_1850": "${input_dir}/${echam.resolution}/New_Hampshire_LCC/hist${jsbachnc_file_ending}/LUH_scaled_transitions_${echam.resolution}_1850.nc"
```

# Summary

If we decide to change the default jsbach landuseTransition forcing file from zero to the 1850 file for the piControl experiment in the esm_tools (which I think we should), the piControl state will change if jsbach runs with dynamic vegetation. 

The default piControl setup would then also differ from the default mpi-esm piControl setup. It would help to ask a responsible person of mpi why their default option is the zero transitions file. Which person would that be? Who could formulate a mail or whatever?

For further comparison of the zero versus the 1850 transitions file, feel free to check the awicm-1.0-recom output 
```
zero: /work/ba1103/a270073/out/awicm-1.0-recom/awi-esm-1-1-lr_kh800/piControl 
1850: /work/ba1103/a270073/out/awicm-1.0-recom/awi-esm-1-1-lr_kh800/piControl_LUtrans1850
```
with the 50 overlapping piControl filename years 2951 to 3000 (spinup years 1002 to 1051).

Cheers,
Chris

**Comment by pgierz** [comment]
Hi Chris,

thanks for looking into this, we just stumbled across it again with @gknorr. I can add the needed files to the defaults by hand, but having it in the main branch would be good.

One important consideration: does this impact the global vegetation coverage as well? If you switch off the BGC module, the atmospheric $CO_{2}$ should not be influenced as far as I know. However, if you have different vegetation coverage, you might end up with different albedo and thus a different base state...

Could someone quickly check the dominant vegetation fraction if there is already a script? Otherwise I will make a notebook to add to our collection.
